packages/flutter/lib/src/services/binary_messenger.dart

```diff
@@ -6,6 +6,7 @@ import 'dart:async';
 import 'dart:typed_data';
 import 'dart:ui' as ui;
 
+import '../../foundation.dart';
```

Comment:
```dart
import 'package:flutter/foundation.dart';
```

Reply:
Updated to absolute path. But what's the rule for imported path? @goderbauer suggested using relative paths within the same package, but seems not always true?

Reply:
Relative imports within the same layer / directory.  services and foundation are different layers and so are treated as different packages.